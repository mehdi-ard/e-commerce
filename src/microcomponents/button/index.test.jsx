import React from 'react';
import renderer from 'react-test-renderer';
import Button from '.';

it('renders default', () => {
  expect(renderer.create(<Button>Hello Button</Button>).toJSON()).toMatchSnapshot();
});

it('renders primary', () => {
  expect(renderer.create(<Button color="primary">primary</Button>).toJSON()).toMatchSnapshot();
});

it('renders primary with variant', () => {
  expect(
    renderer
      .create(
        <Button color="primary" variant="contained">
          primary with variant
        </Button>
      )
      .toJSON()
  ).toMatchSnapshot();
});
