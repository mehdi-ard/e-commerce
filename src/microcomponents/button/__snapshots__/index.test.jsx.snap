// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root Button-button-1  MuiButton-text"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Hello Button
  </span>
</button>
`;

exports[`renders primary 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root Button-button-1  MuiButton-text MuiButton-textPrimary"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    primary
  </span>
</button>
`;

exports[`renders primary with variant 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root Button-button-1  MuiButton-contained MuiButton-containedPrimary"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    primary with variant
  </span>
</button>
`;
